integration-testing/src/main/java/io/grpc/testing/integration/TestServiceImpl.java

```diff
@@ -293,7 +294,14 @@ private void scheduleNextChunk() {
         Chunk nextChunk = chunks.peek();
         if (nextChunk != null) {
           scheduled = true;
-          executor.schedule(dispatchTask, nextChunk.delayMicroseconds, TimeUnit.MICROSECONDS);
+          if (executor instanceof ScheduledExecutorService) {
```

Comment:
With my suggestion we still construct with a ScheduledExecutorService, but we always get the benefit of direct when delaying is unnecessary. That seems like a win overall.